int/hwsc-file-transaction-svc/hwsc-file-transaction-svc.js

```diff
@@ -20,41 +21,61 @@ function getStatus(callback) {
     console.error('callback not a function');
     return;
   }
-
   const request = {};
-
   client.getStatus(request, (err, response) => {
     if (!err) {
       grpc.closeClient(client);
     }
-
     callback(err, response);
   });
 }
 
-function uploadFile(callback) {
+function uploadFile(filePath, fileName, callback) {
+  const fileLocation = filePath + '/' + fileName;
+
   if (typeof callback !== 'function') {
     console.error('callback not a function');
     return;
   }
-  // 'utf8'
-  fs.readFile('res/cat.jpg', 'binary', (errFile, contents) => {
-    if (!errFile) {
+
+  // create a connection from client in API-block to server in Pycharm
+  const server = client.uploadFile((err, response) => {
+    if (!err) {
       // grpc.closeClient(client);
     }
-    var server = client.uploadFile((err, response) => {
-      if (!err) {
-        // grpc.closeClient(client);
-      }
-      callback(err, response);
-    });
-
-    var buf = Buffer.from(contents, 'binary');
-    //console.log(buf);
-    server.write({ buffer: buf });
-    server.end();
+    callback(err, response);
   });
+  // client send the upload file name to server
+  server.write({ fileName: fileName });
+
+  // open the file, and read/pipe the first 1024 bytes of the file
+  const readStream = fs.createReadStream(fileLocation, { hightWaterMark: 1024 });
+
+   readStream.on('readable', () => {
+     let chunk;
+     readStream.pipe(server);
+
+     while ((chunk = readStream.read()) !== null) {
+       server.write({buffer: chunk});
+       readStream.pipe(server);
+        }
+     })
+
+       //readStream.on('end', (err, response) => {
+       .on('end', (err, response) => {
+         server.end();
+         //   callback(err, response);
+       });
+
+       //readStream.on('error', (err, response) => {
```

Comment:
What does `on error` mean in this context? 
`on error` may not be useful for this test, but you should use this in this test.